rust/src/rfb/rfb.rs

```diff
@@ -100,6 +100,18 @@ pub struct RFBState {
     state: parser::RFBGlobalState
 }
 
+#[inline]
+fn handle_incomplete(input: &[u8], current: &[u8], nom_needed: usize) -> AppLayerResult {
```

Comment:
if this `nom_needed` would take `nom::Needed` you could reduce the code at each of the callers.

They could simply be:
```rust
Err(nom::Err::Incomplete(v)) => {
   return handle_incomplete(input, current, v);
}
```
The `if let nom::Needed::Size(n) = v {` could be done once here in `handle_incomplete'
```

Reply:
True. Will look into it.